File: C/mergesort/deepseek-r1-distill-llama-70b/deepseek-r1-distill-llama-70b.c
#include <stdlib.h>
#include <time.h>
#include <stdio.h>

void merge(int *array, int *left, int *right, int n1, int n2) {
    int i = 0, j = 0, k = 0;
    while (i < n1 && j < n2) {
        if (left[i] <= right[j]) {
            array[k++] = left[i++];
        } else {
            array[k++] = right[j++];
        }
    }
    while (i < n1) {
        array[k++] = left[i++];
    }
    while (j < n2) {
        array[k++] = right[j++];
    }
}

void merge_sort(int *array, int size) {
    if (size <= 1) return;
    int mid = size / 2;
    int *left = array;
    int *right = array + mid;
    int n1 = mid, n2 = size - mid;
    merge_sort(left, n1);
    merge_sort(right, n2);
    merge(array, left, right, n1, n2);
}

int main() {
    srand(time(NULL));
    int array[1000];
    for (int i = 0; i < 1000; i++) {
        array[i] = rand() % 1001;
    }
    merge_sort(array, 1000);
    return 0;
}

File: Python/mergesort/deepseek-r1-distill-llama-70b/deepseek-r1-distill-llama-70b.py
def merge_sort(arr):
    """
    Sorts an array using the Merge Sort algorithm.
    
    Args:
        arr (list): The list of elements to be sorted.
    
    Returns:
        list: A new list containing all elements of 'arr' in ascending order.
    """
    # Base case: If the array has 1 or 0 elements, it's already sorted.
    if len(arr) <= 1:
        return arr
    
    # Find the middle point and split the array into two halves.
    mid = len(arr) // 2
    left_half = arr[:mid]
    right_half = arr[mid:]
    
    # Recursively sort the left and right halves.
    left = merge_sort(left_half)
    right = merge_sort(right_half)
    
    # Merge the sorted halves and return the result.
    return merge(left, right)


def merge(left, right):
    """
    Merges two sorted lists into one sorted list.
    
    Args:
        left (list): The first sorted list.
        right (list): The second sorted list.
    
    Returns:
        list: A new list containing all elements from 'left' and 'right' in ascending order.
    """
    merged = []
    left_index = 0
    right_index = 0
    
    # Merge smaller elements first.
    while left_index < len(left) and right_index < len(right):
        if left[left_index] <= right[right_index]:
            merged.append(left[left_index])
            left_index += 1
        else:
            merged.append(right[right_index])
            right_index += 1
    
    # If there are remaining elements in either the left or right, append them.
    merged.extend(left[left_index:])
    merged.extend(right[right_index:])
    
    return merged


import random

def main():
    arr = [random.randint(0, 10000) for _ in range(1000)]
    merge_sort(arr)

if __name__ == "__main__":
    main()
